### PR TITLE
walls internal solution

### PR DESCRIPTION
**Pull request by SansPapyrus683**
someone tell me if the complexity is right all i know is that it works
also i hate java so no java sol
(i could do it, it just would take so long given that there's no pair class and it needs a hashmap)

### CODE REVIEW

File: solutions/ioi-00-walls.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+title: Walls
+author: Kevin Sheng
+---
+
+## Explanation
+
+Instead of treating the towns as nodes, we're going to treat the regions themselves as nodes instead, with
+the walls still being the edges.
+We BFS from each node, logging the minimum distance from every region to every other region.
+Finally, we try all meeting regions and print out the one that minimizes the walls the memebers have to cross.
```
Comment (ar492) [inline solutions/ioi-00-walls.mdx:13]:
```suggestion
Finally, we try all meeting regions and print out the one that minimizes the walls the members have to cross.
```

**Review by ar492 — CHANGES_REQUESTED** [review]

File: solutions/ioi-00-walls.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+</CPPSection>
+<PySection>
+
+```py
+region_num = int(input())
+town_num = int(input())
+member_num = int(input())
+members = [int(i) - 1 for i in input().split()]
+assert member_num == len(members)
```
Comment (ryanchou-dev) [inline solutions/ioi-00-walls.mdx:130]:
```suggestion
```
forgot to remove?

File: solutions/ioi-00-walls.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	for t in range(border_num):
+		regions[r].append((border_towns[t], border_towns[(t + 1) % border_num]))
+
+# the walls previously processed (along with their associated regions)
+prev_walls = {}
+neighbors = [[] for _ in range(region_num)]
+for r in range(region_num):
+	for w in regions[r]:
+		if w in prev_walls:
+			# dupes might hppen but that doesn't really matter
```
Comment (ryanchou-dev) [inline solutions/ioi-00-walls.mdx:150]:
```suggestion
			# dupes might happen but that doesn't really matter
```

**Review by ryanchou-dev — APPROVED** [review]
geniosity :D

File: solutions/ioi-00-walls.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/ioi-00-walls.mdx:130]: i mean no harm in it being there

File: solutions/ioi-00-walls.mdx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+title: Walls
+author: Kevin Sheng
+---
+
+## Explanation
+
+Instead of treating the towns as nodes, we're going to treat the regions themselves as nodes instead, with
+the walls still being the edges.
+We BFS from each node, logging the minimum distance from every region to every other region.
+Finally, we try all meeting regions and print out the one that minimizes the walls the members have to cross.
```
Comment (nathangong) [inline solutions/ioi-00-walls.mdx:13]:
```suggestion
Finally, we try all meeting regions and print out the one that minimizes the number of walls the members have to cross.
```

File: solutions/ioi-00-walls.mdx (hunk 2 above)
Comment (nwatx) [reply solutions/ioi-00-walls.mdx:130]: i think this is ok

**Review by nwatx — CHANGES_REQUESTED** [review]
it seems like some of the comments could potentially be put in the editorial

**Comment by SansPapyrus683** [comment]
what's "some"

**Comment by nwatx** [comment]
> if we've seen this wall before, add that associated region

> dupes might not happen but they don't really matter

probably explaining this would be helpful (if it's not clear already)

**Comment by SansPapyrus683** [comment]
1. how would i explain this in the editorial
2. it's just an implementation thing, doesn't really matter

**Review by nwatx — APPROVED** [review]

**Review by TruaShamu — APPROVED** [review]

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**